inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/deconvolution/deconvolution_kernel_b_fs_zyx_fsv16.cpp

```diff
@@ -155,10 +158,9 @@ JitConstants DeconvolutionKernel_b_fs_zyx_fsv16::GetJitConstants(const deconvolu
     }
     jit.AddConstant(MakeJitConstant("OC_BLOCK", 16));
 
-    if (output.GetDType() == Datatype::F32)
-        jit.AddConstant(MakeJitConstant("DT_F32", 1));
-    else
+    if (output.GetDType() != Datatype::F32) {
```

Comment:
What if you have FP32 -> INT8 deconvolution? I think this check will be invalid for this case.
So probably you should check `inputs[0].GetDType()` instead of `output.GetDType()`

Reply:
>> What if you have FP32 -> INT8 deconvolution?
Do not we want deconvolution to be computed in FP16 precision in this case?

Reply:
No. FP32 input must use FP32 accumulator to avoid accuracy drops in some cases.

Reply:
Changed accordingly, please review.